utils/aws_api.py

```diff
@@ -0,0 +1,306 @@
+                    has_owner = True
+                    break
+        return has_owner
+
+    def custom_s3_filter(self, account, s3, buckets):
+        type = 's3 bucket'
+        unfiltered_buckets = []
+        for b in buckets:
+            try:
+                tags = s3.get_bucket_tagging(Bucket=b)
+            except botocore.exceptions.ClientError:
+                tags = {}
+            if not self.should_filter(account, type, b, tags, 'TagSet'):
+                unfiltered_buckets.append(b)
+
+        return unfiltered_buckets
+
+    def custom_sqs_filter(self, account, sqs, queues):
+        type = 'sqs queue'
+        unfiltered_queues = []
+        for q in queues:
+            tags = sqs.list_queue_tags(QueueUrl=q)
+            if not self.should_filter(account, type, q, tags, 'Tags'):
+                unfiltered_queues.append(q)
+
+        return unfiltered_queues
+
+    def custom_dynamodb_filter(self, account, session, dynamodb, tables):
+        type = 'dynamodb table'
+        dynamodb_resource = session.resource('dynamodb')
+        unfiltered_tables = []
+        for t in tables:
+            table_arn = dynamodb_resource.Table(t).table_arn
+            tags = dynamodb.list_tags_of_resource(ResourceArn=table_arn)
+            if not self.should_filter(account, type, t, tags, 'Tags'):
+                unfiltered_tables.append(t)
+
+        return unfiltered_tables
+
+    def custom_rds_filter(self, account, rds, instances):
+        type = 'rds instance'
+        unfiltered_instances = []
+        for i in instances:
+            instance = rds.describe_db_instances(DBInstanceIdentifier=i)
+            instance_arn = instance['DBInstances'][0]['DBInstanceArn']
+            tags = rds.list_tags_for_resource(ResourceName=instance_arn)
+            if not self.should_filter(account, type, i, tags, 'TagList'):
+                unfiltered_instances.append(i)
+
+        return unfiltered_instances
+
+    def should_filter(self, account, resource_type,
+                      resource_name, resource_tags, tags_key):
+        if self.resource_has_special_name(account, resource_type,
+                                          resource_name):
+            return True
+        if tags_key in resource_tags:
+            tags = resource_tags[tags_key]
+            if self.resource_has_special_tags(account, resource_type,
+                                              resource_name, tags):
+                return True
+
+        return False
+
+    def resource_has_special_name(self, account, type, resource):
+        skip_msg = '[{}] skipping {} '.format(account, type) + \
+            '({} related) {}'
+
+        ignore_names = {
+            'production': ['prod'],
+            'stage': ['stage', 'staging'],
+            'terraform': ['terraform', '-tf-'],
+        }
+
+        for msg, tags in ignore_names.items():
+            for tag in tags:
+                if tag in resource.lower():
```

Comment:
tag.lower() maybe?

Reply:
tags are already lower, but for future tags and forgetfulness :)